base/bucket_n1ql.go

```diff
@@ -72,7 +72,7 @@ func (bucket *CouchbaseBucketGoCB) Query(statement string, params interface{}, c
 		}
 
 		// Non-retry error - return
-		if !isIndexerError(queryErr) {
+		if !isIndexerError(queryErr) && !isSelectBucketError(queryErr) {
```

Comment:
We could add an extra `else if` case into `isIndexerError` which calls out to `isSelectBucketError` instead of adding another explicit check here.

Anywhere else we'd want to know about retryable indexer error, would benefit from the same error checking.

Reply:
Make sense. Added an extra else if case into isIndexerError which calls out to isSelectBucketError.